docs/source/nlp/usage_guides/nlp_embeddings.rst

```diff
@@ -0,0 +1,129 @@
+.. _nlp__embeddings_guide:
+
+=================
+NLP Embeddings
+=================
+
+Embeddings are a way to represent text as a vector of numbers. The vector is a representation of the text in the latent
+space (or semantic space), in which text with similar meaning is represented by similar vectors.
+
+Embeddings are usually extracted from the last layer of a trained neural network model. This model can either be a
+model that was trained on the specific task at hand (e.g. sentiment analysis), or a model that was trained on a
+different task, but is known to be good at extracting embeddings (e.g. BERT).
```

Comment:
I would not use "BERT" as it's a kind of model that can be trained for a narrow task such as sentiment analysis. What about mentioning a specific pre-trained check point that was trained on a general task? Or even mention GPT, why not..

Reply:
I'm not sure I'm following why not BERT.
Also, would prefer not to mention something that is not open source such as GPT.

Reply:
true about OSS
About BERT - my issue is that it's like saying "for embeddings you can either use a model trained for a specific task, or a multilayer perceptron". BERT is just an architecture, than is trained for a certain task. Like you are aiming at here - training some models on some tasks can result in embeddings that generalize, but that a result of the task (e.g. "predict the next word of the text") and the dataset (e.g. "a lot of the internet"). That's why I'm saying that as an example for that you don't want to point at an architecture, but rather at an example for a famous foundation model. An example for that are the BERT models trained as described in https://arxiv.org/abs/1908.10084.